src/deploy/functions/runtimes/index.ts

```diff
@@ -100,36 +98,34 @@ export interface RuntimeDelegate {
   ): Promise<backend.Backend>;
 }
 
-type Factory = (context: args.Context, options: Options) => Promise<RuntimeDelegate | undefined>;
+export interface DelegateContext {
+  projectId: string;
+  // Absolute path of the Firebase project directory.
+  projectDir: string;
+  // Absolute path of the source directory.
+  sourceDir: string;
+  runtime: string;
```

Comment:
nit: I might call this a runtime override or something. This overrules the normal sniffing we do.

Reply:
Also, shouldn't it be optional since the firebase.json value is optional?